src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts

```diff
@@ -17,37 +17,28 @@ import { AuthStorageService } from './auth-storage.service';
 import { NotificationService } from './notification.service';
 
 @Injectable()
-export class AuthInterceptorService implements HttpInterceptor {
-  constructor(
-    private router: Router,
-    private authStorageService: AuthStorageService,
-    public notificationService: NotificationService
-  ) {}
+export class ApiInterceptorService implements HttpInterceptor {
 
-  _notify (resp) {
-    this.notificationService.show(
-      NotificationType.error,
-      resp.error.detail || '',
-      `${resp.status} - ${resp.statusText}`
-    );
-  }
+  constructor(private router: Router,
+              private authStorageService: AuthStorageService,
+              public notificationService: NotificationService) {}
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     return next.handle(request).catch(resp => {
       if (resp instanceof HttpErrorResponse) {
         switch (resp.status) {
-          case 404:
-            this.router.navigate(['/404']);
-            break;
           case 401:
             this.authStorageService.remove();
             this.router.navigate(['/login']);
-            this._notify(resp);
             break;
-          case 500:
-            this._notify(resp);
+          case 404:
+            this.router.navigate(['/404']);
             break;
         }
+        this.notificationService.show(
```

Comment:
We cannot display a generic error when we receive a 409 error code.

409 is used on RBD management to display validation errors returned by ceph.

Reply:
Same problem with RGW; it returns 404 for some errors which is totally wrong in case of the UI because it will redirect the UI to the 404 page. I fixed that by [catching all exceptions](https://github.com/ceph/ceph/pull/21351/files#diff-e695860ccdd34180c4a6dabec48d4490R82) coming from the RGW proxy and return them as 500.

Reply:
Could we extend this to use something like `preventDefault` to prevent the routing in specific cases?

Reply:
@Devp00l Could you please explain it more detailed because this PR does not change general logic of the existing code. maybe your suggestion should be done with a separate PR.

Reply:
@votdev we cannot implement an equivalent solution for RBDs because returning 409 in RBD is intentional.

Reply:
@ricardoasmarques HTTP 409 error codes do [not show notifications anymore](https://github.com/ceph/ceph/pull/21386/commits/cb525d84ede5bd7a26b8df1a4b3f5fd1d76d1d53).